### ISSUE TITLE
Add support for other iCade-like controllers (Submit your info here)

### ISSUE THREAD

**Issue by lmmenge** [lmmenge/MeSNEmu#4]
Many iCade-like controllers (controllers that connect to iOS devices as if they were bluetooth keyboards) have different key maps for their buttons. Take for instance the iCade 8-Bitty: it's different enough from the original iCade that some games are nearly unplayable on this controller.

So in order to add support to various iCade-like controllers, I need your help: if you have an iCade-like controller that is not currently supported by MeSNEmu and you want it supported, please follow these instructions and support for it will be added ASAP:
- Pair your bluetooth controller with an iOS device;
- Make sure your device's keyboard layout is set to US (QWERTY);
- Open Notes;
- Enter the following button sequence matching, as best as you can, the SNES button layout: Up, Right, Down, Left, Select, Start, Y, B, X, A, L, R;
- Copy the text generated (there should be 24 letters) and comment it here along with the controller make and model.

The button sequence you press is very important, since some controllers don't have the exact same layout as a SNES controller. Please try to match it as closely as possible for the best gameplay experience. Below is a picture of a SNES controller. The L and R buttons are on the shoulders of the controller:
![snes-controller](https://f.cloud.github.com/assets/51199/1045205/ee77576e-1020-11e3-8a15-31ff50850e56.png)

**Comment by no2sc2** [comment]
I know the 8bitdo NES30 key map is given but here is my strings. They said the native mode for the 8bitdo NES30 is mode 2. I had problem using mode 2 on my iPhone 6. It wouldn't response for some reason even after chosen the correct controller in the app settings. I had to boot up on mode 3 for it to work. Here are my strings for the 8bitdo nes30.

8bitdo NES30 Mode 2 (as player one "Start+B" boot)
cfdenoijhgkm

Mode 2 (as player two "Start+B+R" boot)
The strings here was as it was set up. The up directional button made the cursor go up, the right directional button made the cursor goes right, down directional button made the cursor go down, left directional button made the cursor go left. Select button was a space. The letter buttons yield as it was, "YBXALR".

Mode 3 (as single player "Start+A" boot)
wedcxzaqlvogythrjnufkpim

I did these strings on my iPhone 6. I did not boot the 8Bitdo nes30 on any other modes because it was only designated for iOS on mode 2 and mode 3 only.

**Comment by lmmenge** [comment]
@no2sc2 MeSNEmu supports the NES30 in Mode 3 as of version 1.4.5 (DerekT07's Mode 3 string matches yours).

**Comment by no2sc2** [comment]
Yes Lucas. I spoke to you about 2 player mode not working. I did a string test anyways so you can get an idea of the keys on mode 2. Maybe it'll help in future updates!

**Comment by acowinthecrowd** [comment]
I have an ipega controller (http://www.amazon.com/Generic-Bluetooth-Controller-Wireless-Joystick/dp/B00AR9JEQU)

Its layout is different than a typical SNES controller, an and xy are switched. I have two strings for you. 

This is hitting the x button on the controller as y, since the SNES has a y where the ipega has an x, etc
wedcxzaqjnufythrimkp

This is the key sequence if I follow the instructions exactly, hitting the x where it shows on the controller.
wedcxzaqythrjnufimkp

I hope that makes sense but if not, let me know.

**Comment by lmmenge** [comment]
@acowinthecrowd just to be clear: if you ignore the letters written on your controller and just press the buttons that are in the same positions (spacially) as the real SNES controller, is this the resulting string? wedcxzaqjnufythrimkp

**Comment by acowinthecrowd** [comment]
That's correct, sorry to make it sound so confusing =)

**Comment by Gohlan** [comment]
Hi, the following text is for a snakebyte controller: wedcxzaqlvogythrjnufimkp

**Comment by lmmenge** [comment]
@Gohlan is that for this controller? http://www.snakebyte-europe.com/en/container-produkte/products/android/idroidcon/

**Comment by lmmenge** [comment]
@acowinthecrowd I just noticed your strings are missing 4 letters. Could you confirm which keys you either don't have or your controller typing while you press?

**Comment by Gohlan** [comment]
Hello,

Yes it is.

Emmanuel Lévy | +33 6 3378 6556 | Envoyé à partir de mon Windows Phone
________________________________
De : Lucas Menge<mailto:notifications@github.com>
Envoyé : ‎14/‎01/‎2015 20:56
À : lmmenge/MeSNEmu<mailto:MeSNEmu@noreply.github.com>
Cc : Gohlan<mailto:emmanuel.levy@live.fr>
Objet : Re: [MeSNEmu] Add support for other iCade-like controllers (Submit your info here) (#4)

@Gohlan is that for this controller? http://www.snakebyte-europe.com/en/container-produkte/products/android/idroidcon/

---

**Comment by naldin** [comment]
Text string for iPega PG-9025: wedcxzaqoglvjnufythrimkp
http://www.dx.com/p/ipega-pg-9025-multimedia-bluetooth-controller-black-256612

**Comment by lmmenge** [comment]
Pushed IPEGA PG-9025. Thanks @naldin !
Pushed Snakebyte idroid:con. Thanks @Gohlan !

**Comment by naldin** [comment]
Thanks @Immenge . Do you know when this will be update in http://builds.io?

**Comment by lmmenge** [comment]
@naldin no idea. I don't own/maintain that site. Maybe you should ask its owner directly?

**Comment by naldin** [comment]
Ok @lmmenge I will, thank you again ;)
